CONTRIBUTING.md

```diff
@@ -0,0 +1,217 @@
+* In "Import Project from Maven" Wizard, set followings.
+   * Set "Import Maven Projects Automatically" .
+   * Set Project SDK as Java 1.8
+
+Optionally, you can use the following maven command to build an IDEA project.
+
+```
+mvn idea:idea
+```
+
+See [http://maven.apache.org/plugins/maven-idea-plugin/](http://maven.apache.org/plugins/maven-idea-plugin/) for more information.
+
+##### Useful IDEA Plugins
+
+* ANTLR 4 - https://plugins.jetbrains.com/plugin/7358-antlr-v4-grammar-plugin 
+* Ballerina - https://plugins.jetbrains.com/plugin/9520-ballerina (Need IDEA 2016.3 or newer)
+
+#### Setup Eclipse
+
+##### Importing a Ballerina Project
+
+All Ballerina repositories are developed as maven projects. So you can import any Ballerina project to eclipse similar to any maven project. 
+Navigate to File → Import..
+Select “Existing Maven Projects” under “Maven”
+Upon next, browse the file system, and open the maven module.
+
+##### Installing ANTLR4 plugin
+If you are working with the grammar, it would be useful to have ANTLR 4 IDE plugin installed for eclipse. To install this using eclipse marketplace:
+
+1. Navigate to, Help → Eclipse Marketplace
+
+2. Search for “antlr4”
+
+3. Install ANTLR 4 IDE plugin (which supports antlr 4.x version).
+
+To open the plugin views: 
+
+1. Navigate to: Window → Show View → Other. 
+
+2. Under “ANTLR4” section, select and enable “Parse Tree” view and “Syntax Diagram” views.
+
+### Working With Ballerina Grammar
+
+Ballerina grammar has been implemented using ANTLR4. It is recommended to have some basic understanding on anltr4 grammar syntax and concepts, before working with the Ballerina grammar. “Parr, Terence (January 15, 2013), The Definitive ANTLR 4 Reference” is a good reference to get started with.
+
+Ballerina grammar can be found under docs/grammar in the Ballerina repository. It consists of two files:
+
+* BallerinaLexer.g4 - Contains the lexer rules for Ballerina grammar. Lexer is responsible for tokenizing an input Ballerina source code.
+* BallerinaParser.g4 - Contains the parser rules. Parser listens to the token stream generated by the lexer. High level grammar productions/abstractions are defined in the parser using those tokens.
+
+> TIP: If you want to check and validate a grammar rule you just wrote, you can use the antlr4 plugin of your IDE. See Setting up your development environment section for more details.
+
+#### Generating Parsers
+
+Once a change is done to the any of the grammar files, it is needed to re-generate the parsers. To generate parsers, we first need to download antlr-complete-4.5.3.jar. Then navigate to <ballerina>/docs/grammar, and generate the lexer followed by the parser (order is important) using the following command:
+
+```
+java -jar /home/supun/Downloads/antlr-4.5.3-complete.jar *.g4 -package org.wso2.ballerinalang.compiler.parser.antlr4 -o ../../compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/antlr4/.

```

Comment:
Shall we generify the following path: `/home/supun/Downloads/antlr-4.5.3-complete.jar`?

Reply:
For example, `<path_to_antlr_jar>/antlr-4.5.3-complete.jar`